packages/tools/devtools/devtools-view/src/components/Menu.tsx

```diff
@@ -314,6 +340,15 @@ export interface MenuItemProps {
 	onClick: (event: unknown) => void;
 	text: string;
 	isActive: boolean;
+	/**
+	 * Icon to display next to the container name based on its state.
+	 */
+	stateIcon?: React.ReactElement;
+
+	/**
+	 * Whether the container or container runtime has recent changes.
+	 */
+	hasChanges?: boolean;
```

Comment:
Nit: these should be `readonly`. Component props should really always be `readonly`.
```suggestion
	/**
	 * Icon to display next to the container name based on its state.
	 */
	readonly stateIcon?: React.ReactElement;

	/**
	 * Whether the container or container runtime has recent changes.
	 */
	readonly hasChanges?: boolean;
```

In the future, we should do some cleanup to make sure all input props are readonly in this package. But not now :)